src/pages/authentication/EnterPinPage.tsx

```diff
@@ -30,7 +24,8 @@ const EnterPinPage: React.FC = () => {
 
   const verifyPin = async (id: number, enteredPin: string) => {
     try {
-      const response = await fetch(VERIFY_API, {
+      HEADERS['X-MS-API-ROLE'] = role;
+      const response = await fetch(ENDPOINTS.VERIFY_PIN, {
```

Comment:
These HEADERS['X-MS-API-ROLE'] and role/userContext are so awesome that they work perfectly.